File: benchmark/rf.nim
import ../src/random_forest/random_forest
import ../tests/test_utils
import times

proc iris_bench() =
    let rf = new_random_forest_classifier(100, max_depth=10)
    let X_data = read_X_data("tests/data/X_iris.csv")
    let y_data = read_y_data("tests/data/y_iris.csv")
    rf.fit(X_data, y_data)
    echo "Successfull finished training of a Random Forest of 100 trees"
    let yhat = rf.predict(X_data)
    echo "accuracy train ", accuracy(y_data, yhat)
    
if isMainModule:
    iris_bench()

File: benchmark/rf.py
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
import pandas as pd
from path import Path

def main():
    root = Path(__file__).parent.parent 
    data_folder = root / 'tests' / 'data'
    X_train = pd.read_csv(data_folder / 'X_iris.csv', header=None).values
    y_train = pd.read_csv(data_folder / 'y_iris.csv', header=None).values.ravel()
    rf = RandomForestClassifier(100, max_depth=10)
    rf.fit(X_train, y_train)
    yhat = rf.predict(X_train)
    print('accuracy train', accuracy_score(y_train, yhat))


if __name__ == "__main__":
    main()